src/SimulationRunner.cc

```diff
@@ -194,7 +194,8 @@ SimulationRunner::SimulationRunner(const sdf::World *_world,
   this->node = std::make_unique<transport::Node>(opts);
 
   // TODO(louise) Combine both messages into one.
-  this->node->Advertise("control", &SimulationRunner::OnWorldControl, this);
+  this->node->Advertise("control", &SimulationRunner::OnWorldControlState,
```

Comment:
How about keeping the old `OnWorldControl` for one development cycle? You  can add a warning message to the `OnWorldControl` function indicating that it will be replaced by `OnWorldControlState`.

Reply:
I assume this means we would need two different services then, right? The current service (`control`) that takes a `msgs::WorldControl`, and a new service (should it be called `control/state`?) that takes a `msgs::WorldControlState`.

Reply:
Yeah, that's correct.

Reply:
Both services are supported as of https://github.com/ignitionrobotics/ign-gazebo/pull/1109/commits/3ca5d9dd72626b72209c422464cfb65e92d1eafc and https://github.com/ignitionrobotics/ign-gui/pull/306/commits/7ff16ce47ffbb1e0dc37d005333a4111a13e505d (using the new `control/state` service is the default), so I'll mark this as resolved. If you want to test using the old service (which doesn't take ECM state information), see the discussion in https://github.com/ignitionrobotics/ign-gui/pull/306#discussion_r739588172